apps/admin-portal/src/components/roles/create-role-wizard/role-permisson.tsx

```diff
@@ -0,0 +1,207 @@
+
+    const [ permissionTree, setPermissionTree ] = useState([]);
+    const [ selectedPermissions, setSelectedPermissions ] = useState([]);
+    const [ permissionsOfRole, setPermissionsOfRole ] = useState([]);
+    const [ permissionsLoading, setPermissionsLoading ] = useState(Boolean);
+    const [ collapseAll, setCollapseAll ] = useState(Boolean);
+
+    const {
+        triggerSubmit,
+        onSubmit,
+        role
+    } = props;
+
+    /**
+     * Util method to check permissions already in the selected role.
+     * 
+     * @param permList - Permissions which are available in the selected role
+     * @param nodes - Permmission Node List
+     * @param isParentChecked - Check whether parent is selected
+     */
+    const checkRolePerms = (permList: string[], nodes: any, isParentChecked: boolean): void  => {
+        nodes.forEach(node => {
+            if (permList.includes(node.fullPath)) {
+                node.isChecked = true;
+                if(node.children){
+                    checkRolePerms(permList, node.children, true);
+                }
+            } else if (isParentChecked) {
+                node.isChecked = true;
+                if(node.children){
+                    checkRolePerms(permList, node.children, true);
+                }
+            } else {
+                if(node.children){
+                    checkRolePerms(permList, node.children, false);
+                }
+            }
+        });
+    }
+
+    const getPermsList = (rolePerms?: string[]): void => {
+        getPermissionList().then((response)=> {
+            if (response.status === 200) {
+                const permList = response.data;
+                let permTree: Permission[] = [];
+
+                permTree = permList.reduce((arr, path) => addPath(
+                    path, path.resourcePath.replace(/^\/|\/$/g, "").split('/'), arr,
+                ),[]);
+
+                if (rolePerms) {
+                    checkRolePerms(rolePerms, permTree, false);
+                    setPermissionsOfRole(rolePerms);
+                    setSelectedPermissions(rolePerms)
+                }
+
+                setPermissionTree(permTree);
+                setPermissionsLoading(false);
+            }
+        }).catch(error => {
+            //TODO handle error
+        });
+
+    }
+
+    useEffect(() => {
+        setPermissionsLoading(true);
+
+        if (role) {
+            getPermissionsPerRole(role).then(response => {
+                if (response.status === 200) {
+                    getPermsList(response.data);
+                }
+            }).catch(error => {
+                //TODO: Handle Error
+            })
+        } else {
+            getPermsList();
+        }
+    },[]);
+
+    /**
+     * A util method which will check all the child elements.
+     * 
+     * @param nodes child nodes need to be checked.
+     */
+    const applyCheckStateTo = (nodes: any, checked: boolean): void => {
```

Comment:
Shall we make the declaration more meaningful here? Doesn't get the exact meaning by looking at the variable declaration name.

Reply:
fixed with commit 5fb1030